includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -139,8 +160,8 @@ public function javascript_params() {
 		global $wp;
```

Comment:
I wonder if we can't just inherit the `javascript_params()` function from the parent class and use `$this->javascript_params()` directly?

Reply:
The new UPE gateway does not extend the existing Stripe gateway (`WC_Gateway_Stripe`) as we won't be using any of the methods defined there... it extends directly from `WC_Stripe_Payment_Gateway`.

Right now we are using 4 of the 30+ configs that the other gateway has, if we need more then we can move the common config to a static helper since most of them are either nonces, order data, or calling other static methods.

Reply:
> The new UPE gateway does not extend the existing Stripe gateway (`WC_Gateway_Stripe`) as we won't be using any of the methods defined there... it extends directly from `WC_Stripe_Payment_Gateway`.

Ah, you're right, I didn't realize the inheritance is from the abstract class 🤔 

This does bring up another thought for me though; maybe the `javascript_params()` function should be made part of the abstract `WC_Stripe_Payment_Gateway ` class 🤔. I think it'd certainly serve us well in this case, and it makes sense to have a re-usable function there.

What do you think?

I'm very concerned about repeating the configuration setup here because it's a massive configuration object. If we duplicate the functionality here it will eventually be a nightmare to maintain.

> Right now we are using 4 of the 30+ configs that the other gateway has, if we need more then we can move the common config to a static helper since most of them are either nonces, order data, or calling other static methods.

If this is the case, I wonder if it makes more sense to just create a separate configuration for the UPE class and remove the duplication from the Shortcode `javascript_params()` function?

If you're sure the configuration will be mostly distinct I'd be in favor of that. There's no need to construct a massive configuration object for just 4 properties.

To reduce the duplication we could probably just introduce methods to retrieve the values so we don't have to update them in multiple places if we need to change them in the future? I'm fine with skipping that if the configuration object for UPE is this small though, so I'll leave it to you to decide what makes the most sense here 🙂 

To reduce duplication this is what I'm thinking of:

```php
function javascript_params() {
	return [
		'option_a' => 'value_a',
		'option_b' => 'value_b',
	];
}
```

to

```php
function javascript_params() {
	return [
		'option_a' => $this->get_value_a(), // inherited function
		'option_b' => $this->get_value_b(), // inherited function
	];
}
```

Reply:
Comparing with what we have in WC Payments it doesn't seem like the gateways config have much in common (the same thing happens between `stripe` and `request-button` gateways), so I'm not so sure about having the `javascript_params()` as part of the abstract `WC_Stripe_Payment_Gateway` class, we would only have the `locale` and `ajax_url` there for now 😕... But this is definitely something we should do when/if the config grows.

Reply:
In that case let's just create a brand new `javascript_params()` function for the UPE checkout, with only the configurations necessary for the UPE checkout 🙂